tests/multigrid-global-coarsening/non_nested_multigrid_05.cc

```diff
@@ -0,0 +1,155 @@
+// // ---------------------------------------------------------------------
+// //
+// // Copyright (C) 2020 - 2023 by the deal.II authors
+// //
+// // This file is part of the deal.II library.
+// //
+// // The deal.II library is free software; you can use it, redistribute
+// // it, and/or modify it under the terms of the GNU Lesser General
+// // Public License as published by the Free Software Foundation; either
+// // version 2.1 of the License, or (at your option) any later version.
+// // The full text of the license can be found in the file LICENSE.md at
+// // the top level directory of deal.II.
+// //
+// // ---------------------------------------------------------------------
+
+
+/**
+ * Test global-coarsening multigrid for a uniformly refined mesh both for
+ * simplicial elements and multiple components.
+ */
+
+#include <deal.II/fe/fe_system.h>
+
+#include <deal.II/grid/grid_tools.h>
+
+#include "multigrid_util.h"
+
+template <int dim, int n_components = 1, typename Number = double>
+void
+test(const unsigned int n_refinements,
+     const unsigned int fe_degree_fine,
+     const double       factor)
+{
+  using VectorType = LinearAlgebra::distributed::Vector<Number>;
+
+  const unsigned int min_level = 0;
+  const unsigned int max_level = n_refinements;
+
+  MGLevelObject<Triangulation<dim>>        triangulations(min_level, max_level);
+  MGLevelObject<DoFHandler<dim>>           dof_handlers(min_level, max_level);
+  MGLevelObject<AffineConstraints<Number>> constraints(min_level, max_level);
+  MGLevelObject<std::unique_ptr<Mapping<dim>>> mappings(min_level, max_level);
+  MGLevelObject<std::shared_ptr<MGTwoLevelTransferNonNested<dim, VectorType>>>
+    transfers(min_level, max_level);
+  MGLevelObject<Operator<dim, n_components, Number>> operators(min_level,
+                                                               max_level);
+
+
+  // set up levels
+  for (auto l = min_level; l <= max_level; ++l)
+    {
+      auto &tria        = triangulations[l];
+      auto &dof_handler = dof_handlers[l];
+      auto &constraint  = constraints[l];
+      auto &op          = operators[l];
+
+      std::unique_ptr<FESystem<dim>> fe =
+        std::make_unique<FESystem<dim>>(FE_SimplexP<dim>(fe_degree_fine), dim);
+      std::unique_ptr<Quadrature<dim>> quad =
+        std::make_unique<QGaussSimplex<dim>>(fe_degree_fine + 1);
+      std::unique_ptr<Mapping<dim>> _mapping =
+        std::make_unique<MappingFE<dim>>(FE_SimplexP<dim>(fe_degree_fine));
+
+      mappings[l] = _mapping->clone();
+
+      // set up triangulation
+      Triangulation<dim> tria_tmp;
+      GridGenerator::hyper_cube(tria_tmp, -1.0, 1.0);
+      tria_tmp.refine_global(l);
+
+      GridGenerator::convert_hypercube_to_simplex_mesh(tria_tmp, tria);
+      GridTools::distort_random(factor,
+                                tria,
+                                true,
+                                boost::random::mt19937::default_seed);
```

Comment:
@fdrmrc This does not seem to be robust!?

Reply:
That is deterministic since we are the default_seed of mt19937. We are using the same `distort_random()` configuration also in non_nested_transfer_03, where  output file all the entries are saved.